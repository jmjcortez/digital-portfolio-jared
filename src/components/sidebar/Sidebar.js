import React from 'react';
import { Link } from 'react-router-dom';

import './Sidebar.styles.css';

const Sidebar = ({ showSideBar }) => {
  return (
    <div>
    {
      showSideBar ? (
        <div className="sidebar" data-testid='sidebar'>
          <span className="title">Jared <br/> Cortez</span>
          <Link to='/' className="links">
            <div className='link-text'>illustrations</div>
          </Link>
          <Link to='/contact' className="links">
            <div className='link-text'>contact/insta</div>
          </Link>
          <Link to='/about' className="links">
            <div className='link-text'>about</div>        
          </Link>
          <div className="footer">
            <span className="footer-text">All images copyright Â©</span>
            <span className="footer-text">Jared Cortez, 2022</span>
            <span className="footer-text">All rights reserved</span>
          </div>
        </div>
      ) : (
        <div data-testid='empty-div'></div>
      )
      
    }
    </div>
  )
}

export default Sidebar;