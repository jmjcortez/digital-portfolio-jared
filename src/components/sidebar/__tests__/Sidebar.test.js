import '@testing-library/jest-dom';

import { screen, render, cleanup } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import Sidebar from '../Sidebar';

afterEach(() => {
  cleanup();
})

test('should render gallery if showSideBar is true', () => {
  render(<BrowserRouter><Sidebar showSideBar={true} /></BrowserRouter>);
  const sidebarElement = screen.getByTestId('sidebar');
  expect(sidebarElement).toBeInTheDocument();
});

test('should NOT render gallery if showSideBar is false', () => {
  render(<BrowserRouter><Sidebar showSideBar={false} /></BrowserRouter>);
  const emptyDiv = screen.getByTestId('empty-div');
  expect(emptyDiv).toBeInTheDocument();
});

test('matches snaphshot', () => {
  const tree = renderer.create(<BrowserRouter><Sidebar /></BrowserRouter>);
  expect(tree).toMatchSnapshot();
});