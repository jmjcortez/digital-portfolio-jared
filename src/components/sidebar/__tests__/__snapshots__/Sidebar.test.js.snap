// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snaphshot 1`] = `
<div>
  <div
    data-testid="empty-div"
  />
</div>
`;
